// graphile-worker/dist/config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const cosmiconfig_1 = require("cosmiconfig");
const cosmiconfigResult = cosmiconfig_1.cosmiconfigSync("graphile-worker").search();
const cosmiconfig = cosmiconfigResult === null || cosmiconfigResult === void 0 ? void 0 : cosmiconfigResult.config;
exports.defaults = {
    schema: process.env.GRAPHILE_WORKER_SCHEMA ||
        enforceStringOrUndefined("schema", cosmiconfig === null || cosmiconfig === void 0 ? void 0 : cosmiconfig.schema) ||
        "graphile_worker",
    maxContiguousErrors: enforceNumberOrUndefined("maxContiguousErrors", cosmiconfig === null || cosmiconfig === void 0 ? void 0 : cosmiconfig.maxContiguousErrors) || 10,
    pollInterval: enforceNumberOrUndefined("pollInterval", cosmiconfig === null || cosmiconfig === void 0 ? void 0 : cosmiconfig.pollInterval) || 2000,
    concurrentJobs: enforceNumberOrUndefined("concurrentJobs", cosmiconfig === null || cosmiconfig === void 0 ? void 0 : cosmiconfig.concurrentJobs) ||
        1,
    maxPoolSize: enforceNumberOrUndefined("maxPoolSize", cosmiconfig === null || cosmiconfig === void 0 ? void 0 : cosmiconfig.maxPoolSize) || 10,
};
function enforceStringOrUndefined(keyName, str) {
    if (typeof str === "string") {
        return str;
    }
    else if (!str) {
        return undefined;
    }
    else {
        throw new Error(`Expected '${keyName}' to be a string (or not set), but received ${typeof str}`);
    }
}
function enforceNumberOrUndefined(keyName, nr) {
    if (typeof nr === "number") {
        return nr;
    }
    else if (typeof nr === "string") {
        const val = parseFloat(nr);
        if (isFinite(val)) {
            return val;
        }
        else {
            throw new Error(`Expected '${keyName}' to be a number (or not set), but received ${nr}`);
        }
    }
    else if (!nr) {
        return undefined;
    }
    else {
        throw new Error(`Expected '${keyName}' to be a number (or not set), but received ${typeof nr}`);
    }
}
